frontend/packages/console-plugin-sdk/src/typings/perspectives.ts

```diff
@@ -20,6 +20,8 @@ namespace ExtensionProperties {
     getK8sLandingPageURL: GetLandingPage;
     /** The function to get redirect URL for import flow. */
     getImportRedirectURL: (project: string) => string;
+    /** The hook to detect default perspective */
+    usePerspectiveDetection?: () => boolean; // isDetectedPerspective
```

Comment:
Not a big fan of tri-state booleans. But definitely not without documentation.
Alternatively return a tuple:
`[enablePerspective: boolean, loading: boolean]`

Reply:
Updated.